plugins/modules/dconf.py

```diff
@@ -401,11 +405,23 @@ def main():
         argument_spec=dict(
             state=dict(default='present', choices=['present', 'absent', 'read']),
             key=dict(required=True, type='str', no_log=False),
-            value=dict(required=False, default=None, type='str'),
+            # Converted to str below after special handling of bool.
+            value=dict(required=False, default=None, type='raw'),
         ),
         supports_check_mode=True
     )
 
+    # Try to be forgiving about the user specifying a boolean as the value, or
+    # more accurately about the fact that YAML and Ansible are quite insistent
+    # about converting strings that look like booleans into booleans. Convert
+    # the boolean into a string of the type dconf will understand. Any type for
+    # the value other than boolean is just converted into a string directly.
+    if module.params['value'] is not None:
+        if isinstance(module.params['value'], bool):
+            module.params['value'] = 'true' if module.params['value'] else 'false'
+        else:
+            module.params['value'] = str(module.params['value'])
```

Comment:
Please use
```suggestion
            module.params['value'] = to_native(module.params['value'], errors='surrogate_or_strict')
```
instead. This is what AnsibleModule will do when `type='str'` is in the argument spec.

(You need `from ansible.module_utils.common.text.converters import to_native` for this if it isn't there already.)